interactions/console.py

```diff
@@ -1,52 +1,75 @@
+            a = int(stri)
+            return True
+        except ValueError:
+            return False
+
+    @staticmethod
+    def clear_view():  # czyszczenie ekranu TYLKO NA LINUXS!!!
         os.system('clear')
 
-    def output (text): # Print w funkcji
+    @staticmethod
+    def output(text):  # Print w funkcji
         print(text)
 
-    def input_text (entry): # Zamiana na string i return
-        text = str(entry)
+    @staticmethod
+    def input_text(entry):  # Wzięcie string i return
+        text = input(str(entry))
         return text
 
-    def input_number (stri,rang): # czy jest to liczba w  liście
-        try:
-            numer = int(stri)
-            is_in_range = False
-            for x in rang:
-              if x == numer:
-                    is_in_range = True
-            if is_in_range == True:
-                return numer
+    @staticmethod
+    def input_number(stri, rang):  # Czy jest to liczba w  liście
+        entry = input(stri)
+        while True:
+            if Console.is_int(entry):
+                numer = int(entry)
+                is_in_range = False
+                for x in rang:
+                    if x == numer:
+                        is_in_range = True
+                if is_in_range:
+                    return numer
+                else:
+                    a = ['Ten numer nie nalerzy do listy. Podaj numer należący do listy: ', str(rang)]
+                    a = ''.join(a)
+                    entry = input(a)
+
             else:
-                a = ['Podaj całkowity numer należący do listy: ', str(rang)]
+                a = ['To nie jest numer. Podaj numer należący do listy: ', str(rang)]
                 a = ''.join(a)
-                newn = input(a)
-                Console.input_number(newn, rang)
-        except ValueError:
-            a = ['Podaj całkowity numer należący do listy: ',str(rang) ]
-            a = ''.join(a)
-            newn = input(a)
-            Console.input_number(newn, rang)
+                entry = input(a)
 
-    def input_numer_in_range (stri,beginr,endr): #czy jest to liczba pomiędzy tymi
-        try:
-            numer = int(stri)
-            is_in_range = False
-            for x in range(beginr, endr + 1):  # pomiędzy łacznie z liczbą końcową
-                if x == numer:
-                    is_in_range = True
-            if is_in_range == True:
-                return numer
+    @staticmethod
+    def input_numer_in_range(stri, beginr, endr):  # czy jest to liczba pomiędzy tymi
+        entry = input(stri)
+        while True:
+            if Console.is_int(entry):
+                numer = int(entry)
+                if beginr <= numer <= endr:
+                    return numer
+                else:
+                    a = ['Ten numer nie nalerzy do zakresu. Podaj całkowity numer w zakresie: ', str(beginr), ' do ',
+                         str(endr)]
+                    a = ''.join(a)
+                    entry = input(a)

```

Comment:
tablica 'a' i jeszcze do tego join, to wygląda bardzo nie elegancko. wystarczy wykorzystać f string od razu wewnątrz input()

Reply:
To akurat nie działa, zajęło mi dobre pół godziny aby znaleźć wersję tych 3 linijek która zadziała poprawnie i to jest jedyna które daje porządne efekty

Reply:
nie. nie jest to jedyna wersja która działa. bo f string działa i wygląda lepiej:

![input_num](https://github.com/wikimody/Quarto-project/assets/150141561/9d35940a-041c-42bd-bdaf-058712daaee6)

Reply:
Och w ten sposób. Nie wiedziałem/może zapomniałem że coś takiego istnieje

Reply:
Sprawdziłem i faktycznie działa. Dzięki za przypomnienie mi że ta funkcja istnieje